dotnet-authserver/src/TeacherIdentity.AuthServer/Journeys/CreateUserHelper.cs

```diff
@@ -0,0 +1,139 @@
+using System.Diagnostics;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using TeacherIdentity.AuthServer.Models;
+using TeacherIdentity.AuthServer.Oidc;
+using TeacherIdentity.AuthServer.Services.UserVerification;
+using TeacherIdentity.AuthServer.Services.Zendesk;
+using ZendeskApi.Client.Models;
+
+namespace TeacherIdentity.AuthServer.Journeys;
+
+public class CreateUserHelper
+{
+    private readonly TeacherIdentityServerDbContext _dbContext;
+    private readonly IUserVerificationService _userVerificationService;
+    private readonly IClock _clock;
+    private readonly IZendeskApiWrapper _zendeskApiWrapper;
+
+    public CreateUserHelper(
+        TeacherIdentityServerDbContext dbContext,
+        IUserVerificationService userVerificationService,
+        IClock clock,
+        IZendeskApiWrapper zendeskApiWrapper)
+    {
+        _dbContext = dbContext;
+        _userVerificationService = userVerificationService;
+        _clock = clock;
+        _zendeskApiWrapper = zendeskApiWrapper;
+    }
+
+    public async Task<IActionResult> CreateOrMatchUserWithTrn(SignInJourney journey, string currentStep)
```

Comment:
This is lifted from the old `TrnCreateUserPageModel`. The intention is that we move all user creation logic here, for all journeys.